pkgs/os-specific/linux/cpupower/gui.nix

```
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub
+, desktop-file-utils
+, gettext
+, glib
+, gobject-introspection
+, gtk3
+, hicolor-icon-theme
+, meson
+, ninja
+, pkg-config
+, python3
+, systemd
+, wrapGAppsHook
+}:
+
+let
+  # use pythonEnv, so that patchShebang fixup will choose this interpreter
+  pythonEnv = python3.withPackages(ps: with ps; [
+    pygobject3
+    dbus-python
+  ]);
+in
+
+stdenv.mkDerivation rec {
+  pname = "cpupower-gui";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "vagnum08";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1rrzlcmspill7qivv3bg8v810ppj4wxa9jcf2ddrjqcbq4lfxayk";
+  };
+
+  nativeBuildInputs = [
+    hicolor-icon-theme # needed for postinstall script
+    desktop-file-utils # needed for update-desktop-database
+    meson
+    ninja
+    glib # needed for glib-compile-scemas
+    pkg-config
+    wrapGAppsHook
+    gettext
+    pythonEnv
+    gtk3
+    gobject-introspection # need for gtk namespace to be available
+  ];
+
+  mesonFlags = [
+    "-Dsystemddir=${placeholder "out"}/lib/systemd"
+  ];
+
+  preConfigure = ''
+    patchShebangs ./build-aux/meson/postinstall.py
+  '';
+
+  # Polkit rules aren't installed and we need them to authorize users via polkit
+  # we refer to this file in the corresponding nixos module
+  postInstall = ''
+    mkdir ${placeholder "out"}/share/polkit-1/rules/
+    cp ../data/polkit/cpupower-gui.rules ${placeholder "out"}/share/polkit-1/rules
+  '';
+
+  meta = with lib; {
+    description = "Change the frequency limits of your cpu and its governor";
+    homepage = "https://github.com/vagnum08/cpupower-gui/";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3Plus;
```

https://github.com/vagnum08/cpupower-gui/blob/50bbf4eadbeec4730042e71b17ce41f6afb4715e/cpupower_gui/cpupower-gui.in#L10

or later